test/ffi/booster.jl

```diff
@@ -161,6 +161,37 @@ end
 end
 
 
+@testset "LGBM_BoosterUpdateOneIterCustom" begin
+
+    mymat = randn(1000, 2)
+    labels = randn(1000)
+    numdata = length(labels)
+    dataset = LightGBM.LGBM_DatasetCreateFromMat(mymat, verbosity)
+    LightGBM.LGBM_DatasetSetField(dataset, "label", labels)
+    # default params won't allow this to learn anything from this useless data set (i.e. splitting completes)
+    booster = LightGBM.LGBM_BoosterCreate(dataset, verbosity)
+
+    finished = LightGBM.LGBM_BoosterUpdateOneIterCustom(booster, randn(numdata), rand(numdata))
+    pred1 = LightGBM.LGBM_BoosterGetPredict(booster, 0)
+    # check both types of float work
```

Comment:
is this randn(numdata) type of Vector{<:AbstractFloat}?

Reply:
I think the test would fail with a MethodError if it was not

Reply:
it is typeof Vector{Float64} which is the same as the following test Float32.(randn(numdata))

Reply:
Well it's different -- the following test is Float32

The API takes only Float32 data, so if the user passes Float64, it needs to be converted. I test each way independently just to verify both entry points work as expected.